pkg/adapter/network.go

```diff
@@ -98,7 +94,22 @@ func (r *LocalRuntime) NetworkRemove(ctx context.Context, cli *cliconfig.Network
 		if err != nil {
 			return networkRmSuccesses, networkRmErrors, err
 		}
-		if err := r.removeNetwork(ctx, name, containers, cli.Force); err != nil {
+		// We need to iterate containers looking to see if they belong to the given network
+		for _, c := range containers {
+			if util.StringInSlice(name, c.Config().Networks) {
+				// if user passes force, we nuke containers
+				if cli.Force {
```

Comment:
maybe this could be happy path

```go
if !cli.Force {
        // Without the force option, we return an error
        return nil, nil, errors.Errorf("%q has associated containers with it. use -f to forcibly delete containers", name)
}

if err := r.RemoveContainer(ctx, c.Container, true, true); err != nil {
    return nil, nil, err
}
```